src/from_dict.jl

```diff
@@ -46,6 +46,13 @@ function from_dict(::Type{OptionType}, d::AbstractDict{String}; kw...) where {Op
     return from_dict_specialize(OptionType, d)
 end
 
+"""
+    from_dict(::Type{T}, t::T) where {T}
+
+An already-parsed option should be kept in nested parsing.
```

Comment:
just remove this docstring and change it to comment since this is not part of the interface. rest are good to merge